src/Orleans.Runtime/MembershipService/MembershipTableManager.cs

```diff
@@ -102,6 +103,24 @@ public async Task RefreshFromSnapshot(MembershipTableSnapshot snapshot)
                 await pending;
             }
 
+            this.log.LogInformation("Received cluster membership snapshot via gossip: {Snapshot}", snapshot);
+
+            if (snapshot.Entries.TryGetValue(this.myAddress, out var localSiloEntry))
+            {
+                if (localSiloEntry.Status == SiloStatus.Dead && this.CurrentStatus != SiloStatus.Dead)
+                {
+                    var msg = $"I should be Dead according to membership table (in RefreshFromSnapshot). Local entry: {(localSiloEntry.ToFullString(full: true))}.";
+                    this.log.Warn(ErrorCode.MembershipFoundMyselfDead1, msg);
+                    this.KillMyselfLocally(msg);
+                }
+
+                snapshot = MembershipTableSnapshot.Create(localSiloEntry.WithStatus(this.CurrentStatus), snapshot);
+            }
+            else
+            {
+                snapshot = MembershipTableSnapshot.Create(this.CreateLocalSiloEntry(this.CurrentStatus), snapshot);
+            }
+
```

Comment:
Why modify the membership info?  Shouldn't we act on the authoritative version rather than modifying it?

Reply:
Will attempt to address in follow up PR